include/audio/aseq.h

```diff
@@ -0,0 +1,1979 @@
+.endm
+
+.macro ldptr lbl
+    _wr_cmd_id  ldptr, ,0xCE,,,,,,, 0, 0
+    _wr_lbl     \lbl
+.endm
+
+// ldptr but for immediates instead of labels
+.macro ldptri imm
+    _wr_cmd_id  ldptr, ,0xCE,,,,,,, 0, 0
+    _wr_u16     \imm
+.endm
+
+/**
+ * rand <max:u8>
+ *
+ *  Stores a random number in the range [0, max) into TR. If max is 0 the range is [0, 255]
+ */
+.macro rand max
+    _wr_cmd_id  rand, 0xCE,0xB8,,,,,,, 0, 0
+    _wr_u8      \max
+.endm
+
+.macro dyncall label=-1
+    .if \label == -1
+        _wr_cmd_id  dyncall, ,0xE4,,,,,,, 0, 0
+    .else
+        _wr_cmd_id  dyncall, 0xCD,,,,,,,, 0, 0
+        _wr_lbl     \label
+    .endif
+.endm
+
+/**
+ * ldi <imm:u8>
+ *
+ *  Loads the immediate value `imm` into TR.
+ */
+.macro ldi imm
+    _wr_cmd_id  ldi, 0xCC,0xCC,,,,,,, 0, 0
+    _wr_u8      \imm
+.endm
+
+/**
+ * and <imm:u8>
+ *
+ *  Computes TR = TR & imm
+ */
+.macro and imm
+    _wr_cmd_id  and, 0xC9,0xC9,,,,,,, 0, 0
+    _wr_u8      \imm
+.endm
+
+/**
+ * sub <imm:u8>
+ *
+ *  Computes TR = TR - imm
+ */
+.macro sub imm
+    _wr_cmd_id  sub, 0xC8,0xC8,,,,,,, 0, 0
+    _wr_u8      \imm
+.endm
+
+/**
+ * stseq <imm:u8> <label:lbl>
+ *
+ *  Stores the u8 value `TR + imm` to the location specified by `label`.
+ */
+.macro stseq imm, label
+    _wr_cmd_id  stseq, 0xC7,0xC7,,,,,,, 0, 0
+    _wr_u8      \imm
+    _wr_lbl     \label
+.endm
+
+.macro stop
+    _wr_cmd_id  stop, 0xC6,0xEA,,,,,,, 0, 0
+.endm
+
+/**
+ * font <fontId:u8>
+ *
+ *  Set the current soundfont for this channel to `fontId`.
+ */
+.macro font fontId
+    _wr_cmd_id  font, ,0xC6,,,,,,, 0, 0
+    _wr_u8      \fontId
+.endm
+
+.macro scriptctr arg
+    _wr_cmd_id  scriptctr, 0xC5,,,,,,,, 0, 0
+    _wr_u16     \arg
+.endm
+
+.macro dyntbllookup
+    _wr_cmd_id  dyntbllookup, ,0xC5,,,,,,, 0, 0
+.endm
+
+/**
+ * runseq <seqPlayer:u8> <seqId:u8>
+ *
+ *  Plays the sequence seqId on seqPlayer
+ */
+.macro runseq seqPlayer, seqId
+    _wr_cmd_id  runseq, 0xC4,,,,,,,, 0, 0
+    _wr_u8      \seqPlayer
+    _wr_u8      \seqId
+.endm
+
+#if (MML_VERSION == MML_VERSION_MM)
+
+    .macro mutechan arg0
+        _wr_cmd_id  mutechan, 0xC3,,,,,,,, 0, 0
+        _wr_s16     \arg0
+    .endm
+
+#endif
+
+/**
+ * noshort
+ *
+ *  Disable short notes encoding.
+ */
+.macro noshort
+    _wr_cmd_id  noshort, ,0xC4,,,,,,, 0, 0
+.endm
+
+/**
+ * short
+ *
+ *  Enable short notes encoding.
+ */
+.macro short
+    _wr_cmd_id  short, ,0xC3,,,,,,, 0, 0
+.endm
+
+.macro dyntbl label
+    _wr_cmd_id  dyntbl, ,0xC2,,,,,,, 0, 0
+    _wr_lbl     \label
+.endm
+
+/**
+ * instr <instNum:u8>
+ *
+ *  Set instrument `instNum` from the current soundfont as the active instrument for this channel or layer.
+ */
+.macro instr instNum
+    _wr_cmd_id  instr, ,0xC1,0xC6,,,,,, 0, 0
+    _wr_u8      \instNum
+.endm
+
+#if (MML_VERSION == MML_VERSION_MM)
+
+    .macro unk_BE arg0
+        _wr_cmd_id  unk_BE, ,0xBE,,,,,,, 0, 0
+        _wr_u8      \arg0
+    .endm
+
+#endif
+
+.macro randptr arg0, arg1
```

Comment:
no description?

Reply:
A lot of commands are lacking a description since I didn't look into them all. I'm not sure I want to hold this up until they all get descriptions, but I may do some more.